sycl/doc/extensions/proposed/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -218,9 +218,16 @@ methods on a modifiable graph. Each node represent either a command-group
 function, empty operation, or device memory allocation/free.
 
 | Edge
-| In the explicit graph building API edges are defined by the user. This is
-either through buffer accessors, the `make_edge()` function, or by passing
-dependent nodes on creation of a new node.
+| In the explicit graph building API edges are primarily defined by the user
+through newly added interfaces. This is either using the `make_edge()` function
+to define an edge between existing nodes, or using a
+`property::node::depends_on` property list when adding a new node to the graph.
```

Comment:
What was the motivation to introduce the property? 

An alternative would be to have `command_graph::add` take a `vector` of dependent nodes, similar to how the queue shortcut functions take a list of events. 

Compared to that, the advantage of the property would be to have only a single `property_list` parameter.

Reply:
Motivation was a better self-documenting API and possibility of connecting arbitrary types.

We had support for a `vector` of `node` in an earlier version of the proposal. This aligns with how events are passed but is the least flexible approach. Dependencies would be restricted to objects of type `node` this way. We discussed using a parameter pack for expressing dependencies internally, but found the variant with `properties` best inline with current SYCL spec.

Reply:
Ok, makes sense, thanks for the explanation!